### PR TITLE
add tests for blocking archived users from discord actions

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
Add tests for the feature to block archived users from using discord actions.

Test coverage

image

image

### CODE REVIEW

**Review by iamitprakash — CHANGES_REQUESTED** [review]
in progress.js file #250

const progressDocumentData = { ...progressData, createdAt: createdAtTimestamp, date: progressDateTimestamp };. The problem is that the function is not checking if the date property is already set in the progressData object. If the date property is already set, then the function will overwrite the value with the current date. This could result in the progress document being created with an incorrect date.

**Comment by RitikJaiswal75** [comment]
> in progress.js file https://github.com/Real-Dev-Squad/website-backend/pull/250

> const progressDocumentData = { ...progressData, createdAt: createdAtTimestamp, date: progressDateTimestamp };. The problem is that the function is not checking if the date property is already set in the progressData object. If the date property is already set, then the function will overwrite the value with the current date. This could result in the progress document being created with an incorrect date.

hey @iamitprakash I dont think this comment belongs to this pr.

**Comment by prakashchoudhary07** [comment]
Do we also need to write test for, not blocking active users?

**Comment by RitikJaiswal75** [comment]
> Do we also need to write test for, not blocking active users?

Good point let me add that.

File: test/integration/discord.test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    chai
+      .request(app)
+      .post("/discord-actions/groups")
+      .set("Cookie", `${cookieName}=${jwt}`)
+      .send(roleData)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+        expect(res).to.have.status(403);
```
Comment (heyrandhir) [inline test/integration/discord.test.js:35]: Nit : we can also check the error message.

File: test/integration/discord.test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  it("returns 403 for archived users get method", function (done) {
+    chai
+      .request(app)
+      .get("/discord-actions/groups")
+      .set("Cookie", `${cookieName}=${jwt}`)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+        expect(res).to.have.status(403);
```
Comment (heyrandhir) [inline test/integration/discord.test.js:49]: Nit : we can also check the error message.

**Review by heyrandhir — APPROVED** [review]

File: test/integration/discord.test.js (hunk 2 above)
Comment (RitikJaiswal75) [reply test/integration/discord.test.js]: i dont think it is required as 403 is sent from the middleware

File: test/integration/discord.test.js (hunk 1 above)
Comment (RitikJaiswal75) [reply test/integration/discord.test.js]: i dont think it is required as 403 is sent from the middleware

**Review by iamitprakash — APPROVED** [review]

**Review by heyrandhir — APPROVED** [review]

**Review by heyrandhir — COMMENTED** [review]
Why so many changes are there in yarn.lock file 👀

**Comment by RitikJaiswal75** [comment]
> Why so many changes are there in yarn.lock file eyes

These are integrity changes and the packages have been moved from npmjs to yarnpkgjs

**Review by heyrandhir — DISMISSED** [review]

**Review by heyrandhir — APPROVED** [review]

**Review by prakashchoudhary07 — APPROVED** [review]

**Pull request merged**